agent/api/task/task.go

```diff
@@ -147,6 +149,9 @@ const (
 	sysctlValueOff = "0"
 )
 
+// LogDriverBufferLimit is the value of the user-defined option log-driver-buffer-limit
+var LogDriverBufferLimit = ""
+
```

Comment:
why is this a global variable? I'm new to ECS Agent code as well, but since it serves multiple tasks, and each task might have multiple container each which can have their own buffer limit- a global variable doesn't seem like what we want.

Reply:
because we want set the input in the config but it is in a different package: https://github.com/aws/amazon-ecs-agent/pull/2958/files#diff-0f6cc891620e98230e92868ea887af5e1f11d9e692fc8415a50c1dd0b130cc8cR1289. So I use a global var to store the value from the option.